src/stress_test.cpp

```diff
@@ -17,7 +17,43 @@ std::vector<uint8_t> HexToBytes(const char *hex_proof) {
     return result;
 }
 
-void doit(int thread)
+struct job
+{
+    std::vector<uint8_t> challengebytes;
+    std::vector<uint8_t> inputbytes;
+    std::vector<uint8_t> outputbytes;
+    uint64 number_of_iterations;
+    uint32 discriminant_size;
+    uint8 witness_type;
+};
+
+void doit(int thread, std::vector<job> const& jobs)
+{
+    int cnt = 0;
+    for (job const& j : jobs)
+    {
+        bool const is_valid = CreateDiscriminantAndCheckProofOfTimeNWesolowski(
+            j.challengebytes,
+            j.discriminant_size,
+            j.inputbytes.data(),
+            j.outputbytes.data(),
+            j.outputbytes.size(),
+            j.number_of_iterations,
+            j.witness_type);
+        if (!is_valid) std::terminate();
+/*
+        printf("thread %d cnt %d is valid %d %llu %d\n",
+            thread,
+            cnt,
+            is_valid,
+            j.number_of_iterations,
+            j.witness_type);
+*/
```

Comment:
It's good to have progress I think, plus you need to know if it fails or succeeds just as a sanity check.

Reply:
Ah I see you have std::terminate. Sounds good if it gives you some report?

Reply:
yeah, true. I commented this out because it was really spamming the terminal. presumably slowing down the threads